compiler/front/options.nim

```diff
@@ -1493,19 +1480,23 @@ proc findProjectNimFile*(conf: ConfigRef; pkg: string): string =
   return ""
 
 proc canonicalImportAux*(conf: ConfigRef, file: AbsoluteFile): string =
-  ## Shows the canonical module import, e.g.: system, std/tables,
-  ## fusion/pointers, system/assertions, std/private/asciitables
-  var ret = getRelativePathFromConfigPath(conf, file, isTitle = true)
-  let dir = getNimbleFile(conf, $file).parentDir.AbsoluteDir
-  if not dir.isEmpty:
-    let relPath = relativeTo(file, dir)
-    if not relPath.isEmpty and (ret.isEmpty or relPath.string.len < ret.string.len):
-      ret = relPath
-  if ret.isEmpty:
-    ret = relativeTo(file, conf.projectPath)
-  result = ret.string
+  ## canonical module import filename, e.g.: system.nim, std/tables.nim,
+  ## system/assertions.nim, etc
```

Comment:
Could you document what canonical import paths represent?

From reading the code, I believe that the new meaning is "a module's path relative to its package (or the current project directory)". Is that correct? As for its previous meaning, I don't think I understand what the old code tried to do.

Reply:
I think we arrived at the same solution: https://github.com/nim-works/nimskull/pull/746#discussion_r1223416175